src/SimulationRunner.cc

```diff
@@ -567,10 +570,20 @@ void SimulationRunner::ProcessSystemQueue()
         this->postUpdateStartBarrier->Wait();
         if (this->postUpdateThreadsRunning)
         {
-          system->PostUpdate(this->currentInfo, this->entityCompMgr);
+          auto terminate = this->threadsToTerminate.find(parentEntity);
+          if (terminate != this->threadsToTerminate.end()) {
+            gzdbg << "Terminating thread " << id << ", " << parentEntity <<"\n";
+            this->postUpdateStartBarrier->Drop();
+            this->postUpdateStopBarrier->Drop();
+            break;
+          }
+          thisThreadSystem->PostUpdate(this->currentInfo,
```

Comment:
I ran the test on macOS and reproduced a seg-fault at this line, while the main body of `INTEGRATION_reset_sensors` is at [reset_sensors.cc:305](https://github.com/gazebosim/gz-sim/blob/arjo/feat/remove_systems_with_entities/test/integration/reset_sensors.cc#L305)

I haven't tried to reproduce on 24.04 yet

Reply:
Thanks for spotting this. I'm currently sleuthing around.

Reply:
Fixed. Sorry I had to force push. Just re-worked the whole PR. Its now much cleaner.